evm_loader/program/src/precompile_contracts.rs

```diff
@@ -538,7 +538,7 @@ pub fn ecrecover(
     } else {
         return Capture::Exit((ExitReason::Succeed(evm::ExitSucceed::Returned), vec![0; 32]));
     };
-    let recovery_id = v - 27;
+    let recovery_id = v.wrapping_sub(27);
     let public_key = match secp256k1_recover(&msg[..], recovery_id, &sig[..]) {
```

Comment:
`recovery_id` must be within `[0, 1, 2, 3]`.
Any other values will cause an error in `secp256k1_recover` call.

Correct fix would be to return an error if `v` is outside of `[27, 28, 29, 30]`.
Geth hides this in a crypto library.

Reply:
Yes, that's right. The code will end up returning an error because recovery_id is out of range. Before this patch though the code will panic if recovery_id is out of range.